### ISSUE TITLE
How to start Insightface?

### ISSUE THREAD

**Issue by ELERKSOSY** [deepinsight/insightface#1735]
Hello,

I would like start testing and using _insightface_ for face recognition. At the main repository page no tutorial or guideline and there are many external links for other repositories and pages. This confuse me a lot.

Simply I look up to use this project announced "_InsightFace 2D&3D deep face analysis toolbox, mainly based on PyTorch and MXNet._"  may you advise me please from where I can start? I confirm that I look up for face recognition project showing all details for testing and implementation in a clear way.

Thank you in advance.

**Comment by nttstar** [comment]
Hi you can start with our python package:
https://github.com/deepinsight/insightface/tree/master/python-package

**Comment by ELERKSOSY** [comment]
I have downloaded and tested the Call Detection Models successfully.

After this as I understood for starting the face recognition then I need to train my own model with my own dataset of images then I convert the trained model to ONNX format.

Can you please share with me the link where I can start the face recognition training model and test/inference?

**Comment by ELERKSOSY** [comment]
Any update?

**Comment by nttstar** [comment]
it's hard to tell how to start with face recognition training, maybe we need a book.

**Comment by ELERKSOSY** [comment]
Indeed, to run such a project requires book. For this to clear the ambiguity often for popular projects on Github, it can be found a good guide line and tutorial takes you to run and use the project successfully. Otherwise in most cases it is lost.

However I ran successfully projects should be based on insightface such as [CompreFace](https://github.com/exadel-inc/CompreFace) and [Face-Recognition-with-InsightFace](https://github.com/tuna-date/Face-Recognition-with-InsightFace)

The accuracy for face recognition was low and not as expected. And project like CompreFace it could not detect faces from the grayscale images.

For this reason I was enthusiastic to test deepinsight/insightface (with my dataset of images) wishing to get better result with it.

**Comment by nttstar** [comment]
Projects like CompreFace only support inference, but no training. I believe the training tutorial will be much longer, as we have no idea about how much users know about the pipeline of face recognition and we should start the tutorial from the beginning.
We have plan to add more documentation/guide-line in recent days, when we are free, thanks for your advice!

**Comment by ELERKSOSY** [comment]
Thank you for the swift reply :)
I have sent you an email. Just to discuss about something different. Your reply will be highly appreciated. Thanks again.

**Comment by leduy99** [comment]
Any update yet? I'm also here because I want to train a face regconizer from my custom dataset but still don't know where to start :D

**Comment by Erfun76** [comment]
> Any update yet? I'm also here because I want to train a face regconizer from my custom dataset but still don't know where to start :D

I'm trying to train a model but there is no proper tutorial.
